dts/arm/nordic/nrf54l15_cpuapp_peripherals.dtsi

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+dppic00: dppic@42000 {
+	compatible = "nordic,nrf-dppic";
+	reg = <0x42000 0x808>;
+	status = "disabled";
+};
+
+spi00: spi@4a000 {
+	/*
+	 * This spi node can be either SPIM or SPIS,
+	 * for the user to pick:
+	 * compatible = "nordic,nrf-spim" or
+	 *              "nordic,nrf-spis".
+	 */
+	compatible = "nordic,nrf-spim";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	reg = <0x4a000 0x1000>;
+	interrupts = <74 NRF_DEFAULT_IRQ_PRIORITY>;
+	max-frequency = <DT_FREQ_M(32)>;
+	easydma-maxcnt-bits = <16>;
+	status = "disabled";
+};
+
+uart00: uart@4a000 {
+	compatible = "nordic,nrf-uarte";
+	reg = <0x4a000 0x1000>;
+	interrupts = <74 NRF_DEFAULT_IRQ_PRIORITY>;
+	status = "disabled";
+};
+
+gpio2: gpio@50400 {
+	compatible = "nordic,nrf-gpio";
+	gpio-controller;
+	reg = <0x50400 0x300>;
+	#gpio-cells = <2>;
+	ngpios = <11>;
+	status = "disabled";
+	port = <2>;
+};
+
+timer00: timer@55000 {
+	compatible = "nordic,nrf-timer";
+	status = "disabled";
+	reg = <0x55000 0x1000>;
+	cc-num = <6>;
+	max-bit-width = <32>;
+	interrupts = <85 NRF_DEFAULT_IRQ_PRIORITY>;
+	prescaler = <0>;
+};
+
+dppic10: dppic@82000 {
+	compatible = "nordic,nrf-dppic";
+	reg = <0x82000 0x808>;
+	status = "disabled";
+};
+
+timer10: timer@85000 {
+	compatible = "nordic,nrf-timer";
+	status = "disabled";
+	reg = <0x85000 0x1000>;
+	cc-num = <8>;
+	max-bit-width = <32>;
+	interrupts = <133 NRF_DEFAULT_IRQ_PRIORITY>;
+	prescaler = <0>;
+};
+
+egu10: egu@87000 {
+	compatible = "nordic,nrf-egu";
+	reg = <0x87000 0x1000>;
+	interrupts = <135 NRF_DEFAULT_IRQ_PRIORITY>;
+	status = "disabled";
+};
+
+radio: radio@8a000 {
+	compatible = "nordic,nrf-radio";
+	reg = <0x8a000 0x1000>;
+	interrupts = <138 NRF_DEFAULT_IRQ_PRIORITY>;
+	status = "disabled";
+	dfe-supported;
+	ieee802154-supported;
+	ble-2mbps-supported;
+	ble-coded-phy-supported;
+
+	ieee802154: ieee802154 {
+		compatible = "nordic,nrf-ieee802154";
+		status = "disabled";
+	};
+};
+
+dppic20: dppic@c2000 {
+	compatible = "nordic,nrf-dppic";
+	reg = <0xc2000 0x808>;
+	status = "disabled";
+};
+
+i2c20: i2c@c6000 {
+	compatible = "nordic,nrf-twim";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	reg = <0xc6000 0x1000>;
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	interrupts = <198 NRF_DEFAULT_IRQ_PRIORITY>;
+	status = "disabled";
```

Comment:
Missing: easydma-maxcnt-bits

Reply:
The same for others twim compatible device declarations.

Reply:
Why is this value in devicetree? It should be taken from MDK.
`TWIMxx_EASYDMA_MAXCNT_MAX` is the define to be used.

Reply:
MDK exposes per-instance defines. Looking at SHIM code, I think it would be enough to change this line:
https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/i2c/i2c_nrfx_twim.c#L422-L423
to `.max_transfer_size = BIT_MASK(TWIM##idx##_EASYDMA_MAXCNT_MAX)`.
Or am I missing something here?

Reply:
There was a long discussion about this topic in the past. The decision taken was that we will move more and more data into DT. Personally - I believe it is better than defining it internally in the driver and taking it out by some macro magic. It gives you full view of the system on DTS level and supports your decision on the specific timer or other device taken to do the job. We have to live with the fact that device management in Zephyr is DTS centric.

Reply:
It is well known that using peripheral indices concatenation is a problem in the Zephyr model because standard instantiation just relies on DT instances. We're the only vendor left with such a custom scheme, but one day we'll need to migrate. Using DT means this tiny bit won't need any fix in the future.

Reply:
@rakons As far as I know, this decision was reversed until we have a solution that automatically generates devicetree from IPS. The main issue here is maintaining the DTS in sync with IPS, as this example shows.

Reply:
Added https://github.com/nrfconnect/sdk-zephyr/pull/1477/commits/97019d8f9ebb2062b7afa30d430ca54bcd9465ef